### PR TITLE
add ability to configure a cause model using YAML

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: proof-of-concept
- *JIRA issue*: [MIC-4056](https://jira.ihme.washington.edu/browse/MIC-4056)
- *Research reference*: 

### Changes and notes
Add ability to configure a cause model using YAML 

### Verification and Testing
Ran parallel simulation and verified that cause model validated

### CODE REVIEW

File: src/vivarium_nih_us_cvd/components/causes/causes.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    SusceptibleState,
+    TransientDiseaseState,
+)
+from vivarium_public_health.utilities import TargetString
+
+from vivarium_nih_us_cvd.constants.paths import CAUSE_RISK_CONFIG
+
+
+class Causes:
+    # todo could modify the component configuration parser plugin
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/components/causes/causes.py:21]: Any Jira needed for this?

File: src/vivarium_nih_us_cvd/components/causes/causes.yaml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+#        transitions:
+#            acute_ischemic_stroke_incidence:
+#                source: "susceptible"
+#                sink: "acute_ischemic_stroke"
+#                data_type: "rate"
+#                get_data_functions:
+#                    incidence_rate: None                        # default
+#                    transition_rate: None                       # default
+#                    remission_rate: None                        # default
+#                    proportion: None                            # default
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/components/causes/causes.yaml:38]: Is it still useful to have this commented block?

**Review by mattkappel — APPROVED** [review]
Just some cleanup comments.

File: src/vivarium_nih_us_cvd/components/causes/causes.py (hunk 1 above)
Comment (rmudambi) [reply src/vivarium_nih_us_cvd/components/causes/causes.py:21]: Yes, good reminder. I'll create a ticket

File: src/vivarium_nih_us_cvd/components/causes/causes.yaml (hunk 2 above)
Comment (rmudambi) [reply src/vivarium_nih_us_cvd/components/causes/causes.yaml:38]: I think it's useful until we have some documentation of what defaults are.

**Review by hussain-jafari — APPROVED** [review]

**Comment by hussain-jafari** [comment]
Our typical runs have results starting at 2023 but I'm also OK with starting at 2021 (which i saw in post processing and in the yaml for custom observers). Was that to keep all observers consistent?

**Comment by rmudambi** [comment]
> Our typical runs have results starting at 2023 but I'm also OK with starting at 2021 (which i saw in post processing and in the yaml for custom observers). Was that to keep all observers consistent?

I didn't change the "real" model spec, so you can decide what to do there.

**Pull request merged**